src/pages/patientView/vafPlot/VAFPlot.tsx

```diff
@@ -0,0 +1,131 @@
+        options.height = isNaN(options.height) ? (((500 + label_dist_to_axis) / 2) - options.margin.top - options.margin.bottom) : options.height;
+        this.state = {
+            show_histogram: this.props.init_show_histogram,
+            show_curve: this.props.init_show_curve,
+            options: options
+        };
+
+        this.toggleShowHistogram = this.toggleShowHistogram.bind(this);
+        this.toggleShowCurve = this.toggleShowCurve.bind(this);
+    }
+
+    componentDidMount() {
+        AlleleFreqPlotMulti(this, this.props.data, this.state.options, this.props.order, this.props.colors, this.props.labels);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        // Hack around React: handle updating here, and return false
+        if (this.histogram_checkbox) {
+            //this.histogram_checkbox.setAttribute("checked", nextState.show_histogram);
+            if (nextState.show_histogram) {
+                $(this.div).find('.viz_hist').show();
+            } else {
+                $(this.div).find('.viz_hist').hide();
+            }
+        }
+        if (this.curve_checkbox) {
+            //this.curve_checkbox.setAttribute("checked", nextState.show_curve);
+            if (nextState.show_curve) {
+                $(this.div).find('.viz_curve').show();
+            } else {
+                $(this.div).find('.viz_curve').hide();
+            }
+        }
+
+        return (nextState.show_histogram !== this.state.show_histogram) || (nextState.show_curve !== this.state.show_curve);
+    }
+
+    toggleShowHistogram(evt) {
+        // setState not necessarily synchronous, so to send the new value to callback we need to do this
+        let new_show_histogram = !this.state.show_histogram;
+        this.setState({show_histogram: new_show_histogram});
+    }
+    toggleShowCurve(evt) {
+        // setState not necessarily synchronous, so to send the new value to callback we need to do this
+        let new_show_curve = !this.state.show_curve;
+        this.setState({show_curve: new_show_curve});
+    }
+
+    render() {
+        let controls = [<label key="histogram-toggle"><input ref={(input) => { this.histogram_checkbox = input; }}
+                                      type="checkbox" checked={ this.state.show_histogram } onChange= { this.toggleShowHistogram }/>histogram</label>,
+                        <label key="curve-toggle"><input ref={(input) => { this.curve_checkbox = input; }}
+                                      type="checkbox" checked={ this.state.show_curve } onChange = { this.toggleShowCurve } />density estimation</label>];
+        return (
+            <div style={{display: "inline"}} ref={(div) => { this.div = div; }}>
+                { ( this.props.show_controls ? controls : [] ) }
+                {( this.props.show_controls ? <br/> : [])}
+            </div>
+        );
+    }
+}
+
+VAFPlot.defaultProps = {
```

Comment:
does this work with typescript?  I think i tried it and was getting ts error.  
I think i found that with ts, default props can be handled with static member of component class:

 public static defaultProps: PageProps = {
        foo: "default"
    };